common/rpc/rpc.go

```diff
@@ -97,6 +99,7 @@ func NewFactory(
 	}
 	f.grpcListener = sync.OnceValue(f.createGRPCListener)
 	f.localFrontendClient = sync.OnceValues(f.createLocalFrontendHTTPClient)
+	f.interNodeGrpcConnections = cache.New(1000, nil)
```

Comment:
1000 seems like enough but maybe it should be configurable just in case?

Reply:
There is a SimpleCache in the common package. I will use that. It does not have a size or eviction logic. That fits better here.

Reply:
I have updated it to use SimpleCache.